backend/fms_core/viewsets/_constants.py

```diff
@@ -114,7 +114,7 @@
 
 _sample_identity_filterset_fields: FiltersetFields = {
     "id": PK_FILTERS,
-    "biosample_id": FK_FILTERS,
+    "biosample__id": FK_FILTERS,
```

Comment:
biosample_id is the name of the fk field biosample__id is the Id on the foreign table ... I am not sure if django does the optimisation itself but join is definitely slower than no join...